src/algebra/char_p/basic.lean

```diff
@@ -428,3 +428,16 @@ begin
 end
 
 end
+
+section prod
+
+variables (S : Type v) [semiring R] [semiring S] (p q : ℕ) [char_p R p] [char_p S q]
+
+/-- The characteristic of the product of rings is the least common multiple of the
+characteristics of the two rings. -/
+protected lemma char_p.prod : char_p (R × S) (nat.lcm p q) :=
```

Comment:
This seems pretty harmless to have as an `instance`

Reply:
I think it could be harmful. If `p = q` then the characteristic of the product is `lcm p p` which is `p`, and you would probably prefer to have `char_p p (R  × S)` as an instance, rather than `char_p (lcm p p) (R × S)`

Reply:
Maybe it's okay actually, because there's not actually a diamond and you could just have both instances.

Reply:
The `p` argument to `char_p` isn't an `out_param`, so someone will never be searching for `char_p (R × S) _` (you have the arguments flipped in your comment). They'll always be searching for either `char_p (R × S) p` or `char_p (R × S) (lcm p q)`, and neither is a match for the other.

Adding another instance for when `p = q` seems like a good idea to me.